aea/helpers/base.py

```diff
@@ -312,6 +312,8 @@ def wrapper(*args: Any, **kwargs: Any) -> Callable:
             try:
                 return fn(*args, **kwargs)
             except Exception as e:  # pylint: disable=broad-except  # pragma: no cover  # generic code
+                if len(args) > 0 and getattr(args[0], "raise_on_try", False):
+                    raise e
```

Comment:
Say in a ledger api, I can now do `ledger_api.raise_on_try = True` and then use it programmatically without exceptions being caught early.